### PR TITLE
Feature/d3 asim 3675

### PR DESCRIPTION
**Pull request by spyrostz**
- Added a dictionary on the unsettled energy deviation as part of the asset_info dict that is reported via the client every market cycle. 
- Fixed bug where the energy deviation was not populated for the LoadExternalMixin and PVExternalMixin but only for the Forecast-related Mixins

### CODE REVIEW

File: src/d3a/models/strategy/settlement/strategy.py
Comment (hannesdiedrich) [inline src/d3a/models/strategy/settlement/strategy.py:213]: As this is d3a-api-client user facing, i suggest to use the `DATE_TIME_FORMAT` instead of the UI format. Thanks!

**Review by hannesdiedrich — COMMENTED** [review]
One comment, once addressed, LGTM

File: src/d3a/models/strategy/settlement/strategy.py
Comment (spyrostz) [reply src/d3a/models/strategy/settlement/strategy.py:213]: will change it thanks!

**Review by hannesdiedrich — DISMISSED** [review]
LGTM

File: .pylintrc
Comment (fievelk) [inline .pylintrc:149]: I think the `too-many-ancestors` warning is useful for new code and shouldn't be completely disabled.

File: src/d3a/models/strategy/external_strategies/__init__.py
Comment (Hassan754) [inline src/d3a/models/strategy/external_strategies/__init__.py:303]: Nit
```suggestion
                "settlement markets")
```

File: src/d3a/models/strategy/external_strategies/__init__.py
Comment (Hassan754) [inline src/d3a/models/strategy/external_strategies/__init__.py:320]: What is expected to happen when a mixed in class doesn't define this `_settlement_market_strategy` ?

**Review by Hassan754 — APPROVED** [review]
LGTM

File: src/d3a/models/strategy/external_strategies/__init__.py
Comment (spyrostz) [reply src/d3a/models/strategy/external_strategies/__init__.py:320]: Exception. However, as discussed offline, this is how mixins work, and that's why we use them only in very limited context (essentially only for classes that inherit from BaseStrategy / BidEnabledStrategy, therefore we know that this member will be part of the concrete class).

File: .pylintrc
Comment (spyrostz) [reply .pylintrc:149]: I disagree on that. Class ancestry is not some nit or small part of the code that can be overlooked (e.g. too-many-arguments) but it is one of the first things that will be checked by the class maintainer. Therefore, if this error is raised, it  is almost guaranteed that it is deliberate and with full knowledge of the implementer, and also extremely explicit to the PR reviewer (I doubt that this will go unnoticed in a PR). Therefore no need for a linter to enforce a rule on that, and litter all mixin classes with the disable-pylint statement.

Comment (fievelk) [reply .pylintrc:149]: I still disagree on this, to be honest. I don't think that all of our current `too-many-ancestors` warnings are caused by deliberate choices that couldn't have been avoided.
Overlooking parts of the code is something that happens constantly when we're in a rush; using a linter is also a way to tell the developer to avoid coding in a hurry (or to add comments to notify readers when this is not possible). Disabling linter warnings on a module basis is ok for me, if I have to choose between the two situations.
So I personally still vote to keep this warning enabled. I'll leave it to the other reviewers anyway, as they might have different opinions.

Comment (spyrostz) [reply .pylintrc:149]: Still not persuaded by the comment. The only occurrences of too-many-ancestors that I have seen in the codebase are the Strategies / ExternalMixin (and I am also not sure about the Market classes, but I doubt that we have reached the limit yet), which is a design decision to implement it this way, therefore will not change. If you find another such occurrence then we can talk more about it, because personally I have not seen any.

Comment (spyrostz) [reply .pylintrc:149]: Warnings that pop up only in cases that we know are OK (which IMO is one of them, as I said this warning is only in one place in the codebase which we know is ok, and if new code is added that violates this we will definitely notice in the PR and make sure that it makes sense or not) are not really helpful, and will only end up polluting our codebase with pylint: disable with no benefit at all. 

In addition to having no benefit of leaving it there, this is contrary to the main goal of Pylint, which is to tailor it to your codebase's needs (that's why it is feasible to disable its warnings in so many ways after all). Essentially Pylint's goal is to not treat all its warnings as errors, but to explicitly select which ones should be treated as errors and which should not according to the codebase. 

BTW this comment is not only specific to this warning. IMO it probably makes sense to disable other warning types too, either because we disagree with them, or because they do not fit to our codebase (for instance, one such candidate is too-broad-exception, but I will leave this discussion for another comment :D )

File: src/d3a/models/strategy/external_strategies/__init__.py
Comment (faizan2590) [inline src/d3a/models/strategy/external_strategies/__init__.py:551]: Is it also due to pylint?

**Review by faizan2590 — DISMISSED** [review]
LGTM

File: src/d3a/models/strategy/external_strategies/__init__.py
Comment (spyrostz) [reply src/d3a/models/strategy/external_strategies/__init__.py:551]: Unfortunately ended up here during the implementation, Pylint does not complain about that. Should not hurt to have here though

**Review by Hassan754 — APPROVED** [review]

File: src/d3a/models/strategy/__init__.py
Comment (fievelk) [inline src/d3a/models/strategy/__init__.py:650]: Nit: I favour the idea of adding this property. It would be great if we could adapt the rest of the codebase (replacing the old `area.spot_market` occurrences with the new property).

File: src/d3a/models/strategy/__init__.py
Comment (fievelk) [inline src/d3a/models/strategy/__init__.py:463]: What would happen if we call `settlement_market_strategy_factory` here as well? Wouldn't it be working the same way?

File: tests/strategies/settlement/test_settlement_strategy.py
Comment (fievelk) [inline tests/strategies/settlement/test_settlement_strategy.py:165]: Nit: this can be moved on top

**Review by fievelk — APPROVED** [review]
Some minor questions, LGTM.

File: src/d3a/models/strategy/__init__.py
Comment (spyrostz) [reply src/d3a/models/strategy/__init__.py:650]: I'll merge it now and open a separate PR for this and the other comment. Heads up though, I will disable pylint for this

**Pull request merged**
